macros/APIRef.ejs

```diff
@@ -158,6 +161,15 @@ function buildSublist(pages, title) {
         });
     }
 
+    // Event pages have a title like "Interface: eventname event" which looks silly
+    // in the sidebar. So for event pages we use the slug, which is supposed to be
+    // in the form "eventname_event", and split off "_event" to leave us (hopefully)
+    // with just "eventname" for the link text.
+    if (hasTag(aPage, 'Event')) {
+        let pageName = aPage.slug.split('/').pop();
+        title = pageName.split('_')[0];
```

Comment:
nit: I'd also recommend moving this block of code up a few lines so it comes right after the initial `var title = ` code.  Also, if you ever end up with another special case for titles, consider breaking all of that out into a separate `getTitleForPage(aPage)` function.

Reply:
That was where I originally had it, except that it's then broken in non-en-US locales by https://github.com/mdn/kumascript/blob/afea01d864b791f838d5b89807880ebe7eb8b679/macros/APIRef.ejs#L158 (I would have missed this without tests).

The logic here is a bit tortured, but it would be a more thoroughgoing change to try to un-torture it, and I'm not sure it's worth it.

Reply:
You're totally right. I just saw that the loop was doing something with translations and didn't look at the loop body. I now understand why it is where it is. Sorry!

Reply:
> If the eventname itself had an underscore in it, then this split() technique would fail

I had thought of this and even had a test for it, but I think I had assumed that in a case where the slug was `thing_onething_event`, then the event name was more likely to be `thing`.

But, either way, this is pathological and I like the more explicit nature of your version so I have updated it (and the test).

Reply:
Fortunately, it seems fairly unlikely there will be any events with underscore in the name, as it's not been done before (at least not for any standardized events).